Documentation porting and versioning

giobart opened the pull request:
This PR includes a porting of the original documentation.
<img width="1390" alt="image" src="https://github.com/user-attachments/assets/a4a4422a-8237-48ab-b12c-df2004b62925">


To do so, it provides a releases selector.
<img width="375" alt="image" src="https://github.com/user-attachments/assets/b6926817-c185-4457-b25d-1f81b4e7c014">

Old wiki releases show an alert banner:
<img width="1382" alt="image" src="https://github.com/user-attachments/assets/6175e99a-2353-4700-a4db-3869f8fb2de3">


The original Accordion documentation is part of an archive directory
![image](https://github.com/user-attachments/assets/5305a9c8-bdfb-4038-b586-b2352ed9c299)

All the archived versions of the wiki can be moved there.

To include a new version of the documentation to the release selector the file `layout/partials/versions.html` must be edited. E.g.:
```
<li><a class="dropdown-item current" aria-current="true" href="/docs/getting-started/oak-environment/high-level-setup-overview/">Latest: Bass (v{{ site.Params.doks.docsVersion }}) &nbsp;&nbsp;</a></li>
<li><hr class="dropdown-divider"></li>
<li><a class="dropdown-item" href="/archive/0.4.301/docs/getstarted/get-started-cluster/">Accordion v0.4.301</a></li>

```

giobart committed:
porting of previous wiki

giobart committed:
porting homepage

Malyuk-A commented:
When I click on the latest link: 
![image](https://github.com/user-attachments/assets/a4bc8775-a52f-4c9f-9c49-67ec6023d32a)

I get to this page: docs/

Is this as intended?

![image](https://github.com/user-attachments/assets/eaea4b34-27f0-40b3-ac47-3f310fcb9370)

Malyuk-A commented:
I have never seen GitHub struggle so much displaying the file changes - this PR has now several times broken my github page  - 5k+ FILE changes - massive bloat

I think I know what the issue is:

![image](https://github.com/user-attachments/assets/4fbe6a87-a8bd-402a-965d-69d7f807bd1c)

We have a folder called fontawesome.
I have worked with it once.

IMO there are ~2 ways to work with it:
Online: easy online include - no need to store any files locally - see https://www.youtube.com/watch?v=sJDyEB1uWCw
Offline: Similar to what we do - but we could cut out everything that we do not need and reduce this by ~99%

AFAIK - we only use fontawesome for a SINGLE icon - a github icon: 
![image](https://github.com/user-attachments/assets/2fdb4ef0-0ffb-44d1-8b82-9faaa7f1e28d)

We could easily replace this icon with a SVG without fontawesome at all - see https://www.youtube.com/watch?v=or7amkb0Pk8

giobart commented:
> > "This PR includes a porting of the original documentation."
> 
> Why only a portion? And what do you mean by portion - what bits are not added to this archive?

a porting not a portion

Mjaethers commented on layouts/archive/single.html:
```diff
@@ -0,0 +1,70 @@
+			<h1>{{ .Title }}</h1>
+
+			<div class="grey-box">
+                <span class="alert-icon">&#9888;</span> This documentation refers to a version of Oakestra which is not the latest. Please refer to <a href="/docs">latest</a>.
```
```suggestion
                <span class="alert-icon">&#9888;</span> This page documents a previous version of Oakestra. Please refer to the <a href="/docs/getting-started/oak-environment/high-level-setup-overview/">latest docs</a> for a current version.
```
I think the phrasing here is a bit off

Mjaethers reviewed (commented):
I agree with @Malyuk-A, this is a bit hard to review on account of the number of files, but in general it looks good

giobart committed:
Update layouts/archive/single.html

Co-authored-by: Matthew Humphreys <ge65yay@tum.de>

giobart committed:
scroll down fix

giobart commented:
TODO:

- Move publications inside Docs
- Remove Blog
- Remove github link in landing page 
- change `archive` to `version`

giobart committed:
removed fonts

giobart committed:
Merge branch 'main' into documentation-versioning

giobart committed:
refactor versions and publications

giobart committed:
fixed page padding

giobart commented:
# Updates:

#### Changed Oakestra text with bw oakestra logo and added publications section:

<img width="475" alt="image" src="https://github.com/user-attachments/assets/55a2dae6-683a-4f5f-b7bb-44131036b3b0">
<img width="475" alt="image" src="https://github.com/user-attachments/assets/42897ff9-6629-4683-beb5-0e85f1ede175">

#### The "publication" section is an auto-generated list of publications
<img width="882" alt="image" src="https://github.com/user-attachments/assets/33db4d68-386b-4900-82f9-09099d95e43e">

The list is generated from the publication files under the publications folder 
<img width="214" alt="image" src="https://github.com/user-attachments/assets/ff5fb51f-a193-4f82-9fab-27c72c3ed89a">

Each publication file simply contains the pub info as follows
```
---
title: "Oakestra: A Lightweight Hierarchical Orchestration Framework for Edge Computing"
authors: "Giovanni Bartolomeo, Mehdi Yosofie, Simon Bäurle, Oliver Haluszczynski, Nitinder Mohan, Jörg Ott"
conference: "USENIX Annual Technical Conference (ATC), 2023"
img: "/pubs/Oakestra-ATC2023.png"
badges: ["/img/usenix-badges/usenixbadges-available.png","/img/usenix-badges/usenixbadges-functional.png","/img/usenix-badges/usenixbadges-reproduced.png"]
pdf: "/pubs/Oakestra-ATC2023.pdf"
doi: "https://www.usenix.org/conference/atc23/presentation/bartolomeo"
summary: ""
draft: false
weight: 204
toc: false
---
```


#### Cleaned landing page

<img width="1391" alt="image" src="https://github.com/user-attachments/assets/b351c5b7-0b8c-4e73-9e6e-abe167de2273">


#### Added simple welcome blogpost in the blog 

<img width="795" alt="image" src="https://github.com/user-attachments/assets/a84b6c9b-5b9d-468c-9aa3-0a332cf19d45">

#### Removed the garbage thousands of fontawesome files

#### Added discord handle in the navbar

<img width="1084" alt="image" src="https://github.com/user-attachments/assets/329b9190-d878-4737-8214-53dabafc8d43">


@Malyuk-A @nitindermohan  @Mjaethers lmk if everything is in order so we can push it.

Malyuk-A commented:
Nitpick / Followup: 

![image](https://github.com/user-attachments/assets/30007743-456b-4643-9bbc-c255e37f7c9e)

How hard is it to center the privacy policy - currently, it is off-center 😅

Malyuk-A commented:
Nitpick / Followup:

The Discord link does not seem to work:

![image](https://github.com/user-attachments/assets/5ea02539-4b98-4ce3-a5f8-38ea47be6229)

giobart commented:
> Nitpick / Followup:
> 
> The Discord link does not seem to work:
> 
> ![image](https://private-user-images.githubusercontent.com/65814168/390841012-5ea02539-4b98-4ce3-a5f8-38ea47be6229.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4MDg3NzUsIm5iZiI6MTczMjgwODQ3NSwicGF0aCI6Ii82NTgxNDE2OC8zOTA4NDEwMTItNWVhMDI1MzktNGI5OC00Y2UzLWE1ZjgtMzhlYTQ3YmU2MjI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI4VDE1NDExNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRiYzI2ZjViY2UyMWNhM2VmYWY0MmIyZWQxY2JhMjBhMzVmYTNhODUxZGI1MzJiYzQwMTk5ZTA0YzAzY2U2YWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4yUXq2eaZb9kTL9tLns0thx_iR3w5lt3L0sMG15NCK0)

That's the one from previous docs, I guess we need to generate a new one then? Is the one in oak website working for you? It's weird but it works for me...

Malyuk-A commented:
The publications are not displayed properly on my end:

![image](https://github.com/user-attachments/assets/a4d4184e-5c05-4002-ade9-e9a149b1277c)
![image](https://github.com/user-attachments/assets/469c70dd-833e-4544-a99c-71e5b4c77a87)


I had a quick look and tried a few quick things with no success - If you want me to investigate further, let me know (this happens for me on Brave and Chrome)

Same goes for the actual paper links:

![image](https://github.com/user-attachments/assets/7e1bfc31-0aa5-462d-9e6e-8eb00b69f178)

The last DOI points to nothing:
![image](https://github.com/user-attachments/assets/eee32ec4-ac58-4a4c-bf4b-381dced3c750)

giobart commented:
> The publications are not displayed properly on my end:

Can you quickly try to stop the server, remove the 'public/' folder and re-start it?

Malyuk-A commented:
> That's the one from previous docs, I guess we need to generate a new one then? Is the one in oak website working for you? It's weird but it works for me...

OK nvm - link is fine when I open it in incognito mode - when I do this in my regular browser mode it does not - I assume this is because I also have my browser linked to discord - so it knows that I am already a member and it goes stupid - that is a discord problem not ours - all good
![image](https://github.com/user-attachments/assets/8edc36a7-c190-413f-8d25-78f621756aee)

Malyuk-A commented:
> > The publications are not displayed properly on my end:
> 
> Can you quickly try to stop the server, remove the 'public/' folder and re-start it?

I did - same issue

P.S: Why does github not support threads in PR comments - in gitlab this is soo much cleaner and easier to follow what is going on :(

giobart committed:
removed title formatting

giobart commented:
> > > The publications are not displayed properly on my end:
> > 
> > 
> > Can you quickly try to stop the server, remove the 'public/' folder and re-start it?
> 
> I did - same issue

Can you check with latest push?

> P.S: Why does github not support threads in PR comments - in gitlab this is soo much cleaner and easier to follow what is going on :(

You're totally right! :(

Malyuk-A commented:
>Can you check with latest push?
Sadly still not there - If you want me to I can Investigate this tomorrow

Malyuk-A commented:
I think this is up for debate:

Nitpick/Personal-taste:

I personally have a strange feeling about the current layout/order of the header elements on the right (icons, release, search, etc.) - it feels off to me

![image](https://github.com/user-attachments/assets/a0756bb4-8887-4217-9e2d-786b0af5be9b)

Here are a couple of other examples:

![image](https://github.com/user-attachments/assets/ab434dce-7afc-4715-9185-740548d0a1a8)
https://kubernetes.io/docs/home/

![image](https://github.com/user-attachments/assets/cd38f18b-9abe-46d4-8fe1-5cc0573f2f49)
https://unikraft.org/


Personally I would do this: 
GetStarted - Releases - GitHub - discord - search - dark/light

giobart commented:
> > Sadly still not there - If you want me to I can Investigate this tomorrow

That's absurd D: I wonder what the heck is going on with the images. Tho its just the images right? How do you run the project?

giobart commented:
> I think this is up for debate:
> 
> Nitpick/Personal-taste:
> 
> I personally have a strange feeling about the current layout/order of the header elements on the right (icons, release, search, etc.) - it feels off to me
> 
> Personally I would do this: GetStarted - Releases - GitHub - discord - search - dark/light

Yup the order can be changed, this is i jsut the default for the theme. We should talk about it.

Mjaethers commented:
Personal taste:
I think the blue logo looks a lot nicer than the black one and also works well in light mode, I think we should always use it.
![image](https://github.com/user-attachments/assets/e2ffe91e-50d3-42a7-bbb1-4d6e681cecea)
![image](https://github.com/user-attachments/assets/dfa86aef-4769-41b9-a42c-305c83e1d02f)

Mjaethers commented:
The prev and next cards on the Accordion docs are all over the place. Is this something worth spending time on fixing right now?

giobart committed:
removed title string transform

giobart committed:
added single layout to improve search result

giobart commented:
> I think this is up for debate:
> 
> Nitpick/Personal-taste:
> 
> I personally have a strange feeling about the current layout/order of the header elements on the right (icons, release, search, etc.) - it feels off to me
>
> Personally I would do this: GetStarted - Releases - GitHub - discord - search - dark/light

Based on discussion, I'll create a draft with the following features:
- releases not showing up but shows only while inside Docs section. 
- Left side: Docs Blog Research
- Right side: Social - search - light/dark - get started
- get started gets replaced with releases when inside Docs.
- remove old docs from search
- Fix font changes

nitindermohan commented:
> Personal taste: I think the blue logo looks a lot nicer than the black one and also works well in light mode, I think we should always use it. ![image](https://private-user-images.githubusercontent.com/61728152/390913177-e2ffe91e-50d3-42a7-bbb1-4d6e681cecea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4Nzc2NTUsIm5iZiI6MTczMjg3NzM1NSwicGF0aCI6Ii82MTcyODE1Mi8zOTA5MTMxNzctZTJmZmU5MWUtNTBkMy00MmE3LWJiYjEtNGQ2ZTY4MWNlY2VhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI5VDEwNDkxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU2YzcyMDdiMDMwMTBhMDU4MGU5NzFiM2QxZjA1YzQ1ODY3OGU2YjBkMWY2MzFlNWJjYTJiYmM1YjZiN2VkZDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TOHtNJe5wuqRx6_teK57Wkn5ihl0HWRGd84eoK1X8OQ) ![image](https://private-user-images.githubusercontent.com/61728152/390913329-dfa86aef-4769-41b9-a42c-305c83e1d02f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4Nzc2NTUsIm5iZiI6MTczMjg3NzM1NSwicGF0aCI6Ii82MTcyODE1Mi8zOTA5MTMzMjktZGZhODZhZWYtNDc2OS00MWI5LWE0MmMtMzA1YzgzZTFkMDJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTI5VDEwNDkxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwODdkZmM2YzQ1NzY3ZWY2M2VjMjdiZjQ2MjJkZjExNDhmNGJkNDhmZjRjZDQxNjgyNWI1OTZlOWI5NTM5MDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Mfx3-mS-IBZYhkkySdKGWWOTIBjR3TKMHCIlbQgjb5c)

I second this!

Malyuk-A commented:
I think the following files can be deleted:

![image](https://github.com/user-attachments/assets/a90ed3b4-bf2f-4a39-9544-8ba512c3aaa5)

- assets/img/avatar_hu7fc94d4596db6a879669a889061e6268_294933_300x0_resize_box_3.png
- content/version/0.4.301/docs/oakestra/node-engine.md
- content/version/0.4.301/docs/oakestra/motivations.md

The following files could maybe be deleted but I am not certain (e.g. I could not find where they get used)
- [assets/dashboard/logo.png](https://github.com/nitindermohan/oak-new-docs/pull/15/files#diff-6d5e5ab83e5227972e24f359bfe7f16d3dad8590511440affce132a9277efc57)
- [assets/img/avatar.png](https://github.com/nitindermohan/oak-new-docs/pull/15/files#diff-994f3eb21fe0aa94bc694529ffc361fba2f566cb2efd31bdab10e9e30c9e82f7)
- assets/network/resolution-seq.png
- content/version/0.4.301/docs/oakestra/README-old.md
- static/css/font-awesome.css
- static/css/font-awesome.min.css
- static/fonts/FontAwesome.otf
- static/fonts/fontawesome-webfont.eot
- static/fonts/fontawesome-webfont.svg
- static/fonts/fontawesome-webfont.ttf
- static/fonts/fontawesome-webfont.woff
- static/fonts/fontawesome-webfont.woff2

Could it be the case that the entire "oakestra" folder in the legacy docs are not used and we could delete it?
![image](https://github.com/user-attachments/assets/7198acd3-adb0-4a83-b80b-731d970cf07e)

Malyuk-A commented:
Besides that - I think everything else looks good
I hope that we can merge this PR quickly because it introduces plenty of (isolated) changes - and many nitpicks we (especially) I have/had can also be easily resolved in smaller targeted follow-up PRs. 🙂

giobart committed:
fixes

giobart commented:
<img width="1437" alt="image" src="https://github.com/user-attachments/assets/40b8f925-9fa0-4080-8112-554e06728a70">
<img width="1437" alt="image" src="https://github.com/user-attachments/assets/411eefc2-1e76-4c95-a456-fd43063ec6f9">

- Fixed logo (now just small icon)
- Changed Icons order
- Releases hidden in home page
- Hidden getstarted inside pages != landing page
- uniform fonts
- removed old docs from search index 
- rename publication sections into research
- fixed releases toggle, now showing selection `latest` only when actually in latest docs 
- removed some garbage

Malyuk-A commented on content/research/_index.md:
```diff
@@ -0,0 +1,22 @@
+---
+title: "Research" # Figure out why title not working
```
Has this been resolved?  - if so the comment is outdated

giobart replied:
```suggestion
title: "Research"
```

Malyuk-A reviewed (approved):
Now it looks a lot cleaner and more professional - Truly great job Giovanni!

There are a few very tiny bits like unnecessary empty lines or commented bits here and there in some HTML files - but to be frank - I would merge this ASAP and if need be tidy up / polish some other time

Ups - I thought I already send this out nmv - now ^^

giobart committed:
Update content/research/_index.md

nitindermohan commented:
A few nitpicks from my end:

1. Subtitle font on front page is too small
<img width="818" alt="Screenshot 2024-12-10 at 17 18 21" src="https://github.com/user-attachments/assets/42bf2e60-987b-49b4-8691-2dbf94c4c795">

2. Footer has a weird spacing at the bottom of the page
<img width="1496" alt="Screenshot 2024-12-10 at 17 18 32" src="https://github.com/user-attachments/assets/178d583c-3c1e-4787-831d-a381f71380a1">

3. Release button does not show anything on click
<img width="159" alt="Screenshot 2024-12-10 at 17 18 42" src="https://github.com/user-attachments/assets/ab94c44a-1716-4e69-b982-426ce71edd5b">

4. DOI button shows up weirdly for publications (sometime in same line as conf name, sometime in next line)
<img width="876" alt="Screenshot 2024-12-10 at 17 19 09" src="https://github.com/user-attachments/assets/b36ee984-da9e-4acc-b304-25156df6623e">
<img width="862" alt="Screenshot 2024-12-10 at 17 19 20" src="https://github.com/user-attachments/assets/b20f81c7-7ec9-4d00-ba9e-d8f46568c99b">

giobart committed:
fixed dropdown

giobart committed:
fix research DOI indentation

giobart committed:
removed bootstrap css

giobart committed:
fix research

giobart commented:
> A few nitpicks from my end:
> 
> 1. Subtitle font on front page is too small

Increased font size

> 2. Footer has a weird spacing at the bottom of the page
> 
> 3. Release button does not show anything on click

These were big ones to solve. At first I was not able to replicate the issue, then I noticed that it only occurs on some browsers. After some tinkering I ended up manually adding bootstrap as a module and loading the bootstrap js partials manually in the navbar. Now it looks like it's working.  Please test it once more @nitindermohan 

>  4. DOI button shows up weirdly for publications (sometime in same line as conf name, sometime in next line)

I think I fixed it. But on the whitepaper the conference line remains blank.

nitindermohan commented:
Getting build error on the branch 
```
Error: command error: failed to load modules: failed to download modules: binary with name "go" not found in PATH
```
It is caused by the go.mod and go.sum files that were pushed in commit https://github.com/nitindermohan/oak-new-docs/pull/15/commits/58605296eb6328fcdc75d4caa8185a9ab99f2fa3. Maybe artifacts of the other website as they still refer to another documentation repo.

EDIT: Works after installing `golang` in the machine. @giobart will add new documentation setup instructions with the GoLang dependency.

nitindermohan commented:
The PR looks good now. Merging it! Thanks all for your contributions :)
